packages/javascript-mutator/src/JavaScriptMutator.ts

```diff
@@ -8,14 +8,33 @@ import copy from './helpers/copy';
 import { NODE_MUTATORS_TOKEN, NodeMutator } from './mutators/NodeMutator';
 
 export class JavaScriptMutator implements Mutator {
-  public static inject = tokens(commonTokens.logger, NODE_MUTATORS_TOKEN);
-  constructor(private readonly log: Logger, private readonly mutators: readonly NodeMutator[]) {}
+  public static inject = tokens(commonTokens.logger, commonTokens.mutatorDescriptor, NODE_MUTATORS_TOKEN);
+  constructor(
+    private readonly log: Logger,
+    private readonly mutatorDescriptor: MutatorDescriptor,
+    private readonly mutators: readonly NodeMutator[]
+  ) {
+    /*
+      Edge case check
+      Check if `mutatorDescriptor` has been passed.
+      If it is not and `mutators` isn't as well, `mutatorDescriptor` is undefined
+      If it is not, but `mutators` is, `mutatorDescriptor` is an array
+    */
+    if (typeof mutatorDescriptor !== 'object' || Array.isArray(mutatorDescriptor)) {
+      this.mutatorDescriptor = {
+        babelPlugins: [],
+        excludedMutations: [],
+        name: ''
+      };
+    }
+  }
 
   public mutate(inputFiles: File[]): Mutant[] {
     const mutants: Mutant[] = [];
+    const babelPlugins = this.mutatorDescriptor.babelPlugins || [];
 
     inputFiles.forEach(file => {
-      const ast = BabelHelper.parse(file.textContent);
+      const ast = BabelHelper.parse(file.textContent, babelPlugins);
```

Comment:
This will probably need a cast here. From a `string []` to `ParserPlugin[]`. 

We should check what happens when we supply a plugin that does not exist. What is the error message Stryker provides?

Reply:
I think babel throw an error itself when can't find plugin